### PR TITLE
chore(tests): update tests across modules and add xhr-wrapper

### PR DESCRIPTION
**Pull request by ragaeeb**
- expand test coverage across modules
- add new tests for xhr-wrapper, dom-download, and probe-scheduler
- enhance fetch-wrapper and text-extraction tests with additional scenarios
- extend conversation-readiness, logs-storage, protocol messages, session-token
- extend runner readiness, runtime-settings, sfe probe lease store, text-utils, and title-resolver tests
- add in-memory fallback storage test and storage-read failure paths
- update test scaffolding to capture console.debug logs consistently

## Summary by CodeRabbit

* **New Features**
  * Added utilities for conversation content validation, title normalization, and JSON string cleaning; probe lease store now selects session-backed storage when available.

* **Tests**
  * Vastly expanded test coverage across fetch/XHR interception, text extraction, download and storage utilities, session tokens, probe scheduling/lease stores, runtime readiness/settings, protocol messages, and edge-case logging/behavior.

### CODE REVIEW

File: utils/sfe/probe-lease-store.test.ts [hunk 1]
```diff
@@ -1,24 +1,166 @@
...
+                },
+            };
+            (globalThis as any).browser = fakeBrowser;
+            try {
+                const store = createProbeLeaseStore();
+                await store.set('testKey', 'testVal');
+                expect(await store.get('testKey')).toBe('testVal');
+            } finally {
+                delete (globalThis as any).browser;
+            }
```
[the remark this answers is not in the document]
Comment (ragaeeb) [reply utils/sfe/probe-lease-store.test.ts]: https://github.com/ragaeeb/blackiya/pull/40/commits/718fe3ba5a8d2dde52b4c7ee3cfee5560f4a1006

File: utils/logs-storage.test.ts [hunk 2]
```diff
@@ -169,6 +169,39 @@ describe('BufferedLogsStorage', () => {
...
+                message: 'in-memory test',
+                context: 'background' as const,
+            };
+            await fallbackStorage.saveLog(entry);
+            const logs = await fallbackStorage.getLogs();
+            expect(logs.length).toBe(1);
+            expect(logs[0].message).toBe('in-memory test');
+            await fallbackStorage.clearLogs();
+            const cleared = await fallbackStorage.getLogs();
+            expect(cleared.length).toBe(0);
```
[the remark this answers is not in the document]
Comment (ragaeeb) [reply utils/logs-storage.test.ts]: https://github.com/ragaeeb/blackiya/pull/40/commits/718fe3ba5a8d2dde52b4c7ee3cfee5560f4a1006
